File: src/indicators/collection/ForexStream.java
package indicators.collection;

//TODO: get rid of all locks, leave on clojure side!
 
public class ForexStream extends PriceStream {
	public String symbol = "";
	public int timeframe = 60; 
	public long headTime = 0;  
	
    public ForexStream(String symbol,int timeframe,long headTime) {
    	this.symbol = symbol;this.timeframe = timeframe;this.headTime = headTime;
    }
	public ForexStream(ForexStream p) {
		super(p.capacity(), p);
        this.symbol = p.symbol;
	    this.timeframe = p.timeframe;
		this.headTime = p.headTime;
	}
	
}

File: src/indicators/ForexStream.scala
package indicators
object ForexStream {
  def create(s:PriceStream,symbol:String,timeframe:Int) = {
    val b = new ForexStream(symbol,timeframe,s.size)
    b.reset(s)
    b
  }
  def create(s:ForexStream) = {
    val b = new ForexStream(s.symbol,s.timeframe,s.size)
    b.reset(s)
    b
  }
}

class ForexStream(val symbol:String,val timeframe:Int,the_capacity:Int=1000) extends PriceStream(the_capacity) {
  var headTime:Long = 0
}
 
